app/src/main/java/dk/scuffed/opengltest/pipeline/stages/TextureStage.kt

```diff
@@ -0,0 +1,75 @@
+package dk.scuffed.opengltest.pipeline.stages
+
+import android.content.Context
+import android.graphics.Bitmap
+import android.graphics.BitmapFactory
+import android.opengl.GLES20
+import android.util.Size
+import dk.scuffed.opengltest.R
+import dk.scuffed.opengltest.gl.*
+import dk.scuffed.opengltest.pipeline.*
+import java.nio.ByteBuffer
+
+
+internal class TextureStage(private val context: Context, private val texture : Bitmap, private val pipeline: Pipeline): GLOutputStage(context, R.raw.vertex_shader, R.raw.texture, pipeline) {
```

Comment:
TextureStage does not have to be a GLOutputStage as its texture can be loaded once and sent to other stages without having to draw it into a framebuffer every frame